jina/flow/base.py

```diff
@@ -893,21 +933,40 @@ def _get_routing_table(self) -> RoutingTable:
                     )
 
         for end, pod in self._pod_nodes.items():
+
             if end == GATEWAY_NAME:
                 end = f'end-{GATEWAY_NAME}'
 
             if pod.head_args.hosts_in_connect is None:
                 pod.head_args.hosts_in_connect = []
 
+            if isinstance(pod, K8sPod):
+                from ..peapods.pods.k8slib import kubernetes_deployment
+
+                end = kubernetes_deployment.to_dns_name(end)
```

Comment:
```suggestion
```

Reply:
I hope I could remove it, but this breaks the routing table

Reply:
And the connectionPool relies on the `dns_name` to identify the other Pods

Reply:
I see